src/methods/widen.js

```diff
@@ -121,6 +122,22 @@ export class WidenImplementation {
   widenBindings(realm: Realm, m1: Bindings, m2: Bindings): Bindings {
     let widen = (b: Binding, v1: void | Value, v2: void | Value) => {
       let result = this.widenValues(realm, v1 || b.value, v2 || b.value);
+      if (result instanceof AbstractValue && result.kind === "widening") {
+        let phiNode = b.phiNode;
+        if (phiNode === undefined) {
+          // Create a temporal location for binding and initialize it with its value at the loop start
```

Comment:
But then you never store it back to `b.phiNode`?

Reply:
Oops.